### DISCUSSION TITLE
How is the F1 for the Aspect Term Extraction Polarity Classification (ATEPC) calculated ?

### DISCUSSION THREAD

**Discussion by mohamedamr13** [yangheng95/PyABSA#291] [Q&A]
In ATEPC, each word in a training or validation example has a tag, either outside of aspect, beginning of aspect, or end of aspect, and a sentiment for each word, where aspects have 3 possible sentiments and non-aspects have some sort of padding. How is the F1 score for this training paradigm calculated. For example, a sentence with more than 1 aspect below 

كانت O -100
الغرفه B-ASP Positive
ممتازه O -100
وكذلك O -100
الموظفون B-ASP -100
وبوفيه B-ASP -100
الافطار I-ASP -100
. O -100
ومع O -100
ذلك O -100
فقد O -100
كانت O -100
وجبه B-ASP -100
العشاء I-ASP -100
في O -100
المطعم O -100
باهظه O -100
الثمن O -100
وغير O -100
مرضيه O -100
. O -100

كانت O -100
الغرفه B-ASP -100
ممتازه O -100
وكذلك O -100
الموظفون B-ASP Positive
وبوفيه B-ASP -100
الافطار I-ASP -100
. O -100
ومع O -100
ذلك O -100
فقد O -100
كانت O -100
وجبه B-ASP -100
العشاء I-ASP -100
في O -100
المطعم O -100
باهظه O -100
الثمن O -100
وغير O -100
مرضيه O -100
. O -100

كانت O -100
الغرفه B-ASP -100
ممتازه O -100
وكذلك O -100
الموظفون B-ASP -100
وبوفيه B-ASP Positive
الافطار I-ASP Positive
. O -100
ومع O -100
ذلك O -100
فقد O -100
كانت O -100
وجبه B-ASP -100
العشاء I-ASP -100
في O -100
المطعم O -100
باهظه O -100
الثمن O -100
وغير O -100
مرضيه O -100
. O -100

كانت O -100
الغرفه B-ASP -100
ممتازه O -100
وكذلك O -100
الموظفون B-ASP -100
وبوفيه B-ASP -100
الافطار I-ASP -100
. O -100
ومع O -100
ذلك O -100
فقد O -100
كانت O -100
وجبه B-ASP Negative
العشاء I-ASP Negative
في O -100
المطعم O -100
باهظه O -100
الثمن O -100
وغير O -100
مرضيه O -100
. O -100

**Comment by yangheng95** [comment]
Depends on the token-level classification report.

**Reply by mohamedamr13** [reply]
@yangheng95 Can you please elaborate on that with an example or a resource ?
